### DISCUSSION TITLE
Metrics calculation on CPU

### DISCUSSION THREAD

**Discussion by jhauret** [Lightning-AI/torchmetrics#888] [Ideas]
Hi,

 I was wondering if someone has implemented a GPU version of `torchmetrics.audio.pesq.PerceptualEvaluationSpeechQuality` and `torchmetrics.audio.stoi.ShortTimeObjectiveIntelligibility` which can only be performed on CPU at the moment. If yes, could you make it available for the lightning community ?

Thanks

**Comment by SkafteNicki** [comment]
Sadly, the PESQ metric is trademarked and therefore cannot be re-implemented on GPU. You can read more here:
https://github.com/ludlows/python-pesq/blob/739759de9dde2d592a6f12b67f3e2875abeecb8f/pesq/dsp.c#L39-L48
STOI could probably be re-implemented in pytorch to gpu accelerate as the implementation seems fairly simple:
https://github.com/mpariente/pystoi/blob/master/pystoi/stoi.py

**Reply by jhauret** [reply]
I finally used https://github.com/asteroid-team/pytorch_stoi and encapsulated with class inheriting from `torchmetrics.metric.Metric`
